pacman/utilities/file_format_converters/convert_to_java_machine.py

```diff
@@ -0,0 +1,137 @@
+    """
+
+    __slots__ = []
+
+    def __call__(self, machine, file_path):
+        """
+        Runs the code to write the machine in Java readable json.
+
+        :param machine: Machine to convert
+        :param file_path: Location to write file to. Warning will overwrite!
+        """
+        progress = ProgressBar(
+            (machine.max_chip_x + 1) * (machine.max_chip_y + 1) + 2,
+            "Converting to JSON machine")
+
+        for chip in machine.chips:
+            if (chip.ip_address is None):
+                s_monitors = chip.n_processors - chip.n_user_processors
+                s_router_entries = chip.router.n_available_multicast_entries
+                s_router_clock_speed = chip.router.clock_speed
+                s_sdram = chip.sdram.size
+                s_virtual = chip.virtual
+                s_tags = chip._tag_ids
+                break
+
+        chip = machine.boot_chip
+        e_monitors = chip.n_processors - chip.n_user_processors
+        e_router_entries = chip.router.n_available_multicast_entries
+        e_router_clock_speed = chip.router.clock_speed
+        e_sdram = chip.sdram.size
+        e_virtual = chip.virtual
+        e_tags = chip._tag_ids
+
+        standardResources = OrderedDict()
+        standardResources["monitors"] = s_monitors
+        standardResources["routerEntries"] = s_router_entries
+        standardResources["routerClockSpeed"] = s_router_clock_speed
+        standardResources["sdram"] = s_sdram
+        standardResources["virtual"] = s_virtual
+        standardResources["tags"] = list(s_tags)
+
+        ethernetResources = OrderedDict()
+        ethernetResources["monitors"] = e_monitors
+        ethernetResources["routerEntries"] = e_router_entries
+        ethernetResources["routerClockSpeed"] = e_router_clock_speed
+        ethernetResources["sdram"] = e_sdram
+        ethernetResources["virtual"] = e_virtual
+        ethernetResources["tags"] = list(e_tags)
+
+        # write basic stuff
+        json_obj = OrderedDict()
+        json_obj["height"] = machine.max_chip_y + 1
+        json_obj["width"] = machine.max_chip_x + 1
+        json_obj["root"] = list((machine.boot_x, machine.boot_y))
+        json_obj["standardResources"] = standardResources
+        json_obj["ethernetResources"] = ethernetResources
+        json_obj["chips"] = []
+
+        # handle chips
+        for chip in machine.chips:
+            details = OrderedDict()
+            details["cores"] = chip.n_processors
+            details["ethernet"] =\
+                [chip.nearest_ethernet_x, chip.nearest_ethernet_x]
+            dead_links = []
+            for link_id in range(0, Router.MAX_LINKS_PER_ROUTER):
+                if not chip.router.is_link(link_id):
+                    dead_links.append(link_id)
+            if len(dead_links) > 0:
+                details["deadLinks"] = dead_links
+
+            exceptions = OrderedDict()
+            if chip.ip_address is not None:
```

Comment:
```suggestion
            if chip.ip_address is not None:
                details['ipAddress'] = chip.ip_address
                resources = ethernetResources
            else:
                resources = standardResources    
            if (chip.n_processors - chip.n_user_processors) != resources["monitors"]:
                exceptions["monitors"] = \
                        chip.n_processors - chip.n_user_processors
            if (chip.router.n_available_multicast_entries !=
                        resources["routerEntries"]):
               exceptions["routerEntries"] = \
                        chip.router.n_available_multicast_entries
                if chip.router.clock_speed != resources["routerClockSpeed"]:
                    exceptions["routerClockSpeed"] = \
                        chip.router.n_available_multicast_entries
                if chip.sdram.size != resources["sdram"]:
                    exceptions["sdram"] = chip.sdram.size
                if chip.virtual != resources["virtual"]:
                    exceptions["virtual"] = chip.virtual
                if chip.tag_ids != resources["tags"]:
                    details["tags"] = list(chip.tag_ids)
```
the reasoning for this to avoid cloned code. 
also note the removal of brackets in like 92 as they are redundant

Reply:
its explicit, but lines 84 to 117 are to be replaced with this